roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,137 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class DockerImageAvailability(OpenShiftCheck):
+    """Check that required Docker images are available.
+
+    This check attempts to pull required docker images and fail if
+    any image is missing or an error occurs during pull.
+    """
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    skopeo_image = "juanvallejo/skopeo:latest"
+
+    registry_prefix = {
+        "origin": "openshift/origin",
+        "openshift-enterprise": "openshift3/ose",
+    }
+
+    def run(self, _, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+        deployment_type = get_var(task_vars, "deployment_type")
+
+        reg_prefix = self.registry_prefix[deployment_type]
+        required_images = self.get_required_images(is_containerized, reg_prefix, openshift_release)
+
+        # append images with qualified image tags to our list of required images.
+        # these are images with a (v0.0.0.0) tag, rather than a standard release
+        # format tag (v0.0). We want to check this set in both containerized and
+        # non-containerized installations.
+        required_images.update(self.qualified_docker_images(reg_prefix, "v" + openshift_image_tag))
+
+        # use docker_image_facts to inspect local images before attempting to use Skopeo
+        required_images = self.inspect_local_images(required_images, task_vars)
+
+        # ensure skopeo image exists or can be pulled and used successfully
+        failed = self.check_skopeo_image(task_vars)
```

Comment:
I think in the Linux/UNIX ecosystem it is more common to look at the exit code. If the exit code is 0 and still stderr is not empty, we should consider it a successful run.

Unless the `docker` client doesn't behave that way, I'd suggest we look at the exit code, and if it is non-zero then we can print the combined stdout+stderr while explaining the failure.

---

To keep us honest, while `docker pull` seems fine for now, it doesn't fix the versioning problem -- one could be running an old version of the check and it would go ahead and pull a newer / potentially incompatible image.